### DISCUSSION TITLE
Broken build

### DISCUSSION THREAD

**Discussion by tbedford** [squidfunk/mkdocs-material#6982] [General]
Hello,

We are getting the following error in building our [docs](https://github.com/quixio/quix-docs):

```
- tutorials/train-and-deploy-ml/train-ml.md
  - quix-streams/build/README.md
ERROR   -  Error building page 'apis/portal-api/http-requests.md': File is not a zip file
Traceback (most recent call last):
🔳 quix-streams
  File "/opt/hostedtoolcache/Python/3.12.2/x64/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
✅ quix-streams (0.51 secs)
             ^^^^^
  File "/opt/hostedtoolcache/Python/3.12.2/x64/lib/python3.12/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.2/x64/lib/python3.12/site-packages/click/core.py", line 10[78](https://github.com/quixio/quix-docs/actions/runs/8466316400/job/23194828660#step:27:79), in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.2/x64/lib/python3.12/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
<snip>
```

Everything was working fine yesterday and nothing has changed in terms of content commits. Today, the docs build fails. **However, everything builds fine locally using either `serve` or `build`.**

We are using latest version of MkDocs and Material Insiders on our build server, and Python 3.12.2 on the server. 

I'm using Python 3.12.2 locally too, and latest MkDocs and Material Insiders.

The file that appears to be generating the error (apis/portal-api/http-requests.md) displays fine when building locally, and I can't see any issues with it.

Any ideas on how to debug?  Suggestions?

Best regards.

**Comment by alexvoss** [comment]
The very end of a very long stack trace of your build shows this originates with an attempt to retrieve fonts from Google Fonts. Perhaps something broke over there?

**Reply by tbedford** [reply]
Thanks for that. For reference here's the end of the stack trace you are referring to:

```
File "/opt/hostedtoolcache/Python/3.12.2/x64/lib/python3.12/site-packages/material/plugins/social/plugin.py", line 522, in _render_typography
    path = self._resolve_font(family, style, variant)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.2/x64/lib/python3.12/site-packages/material/plugins/social/plugin.py", line 809, in _resolve_font
    self._fetch_font_from_google_fonts(family)
  File "/opt/hostedtoolcache/Python/3.12.2/x64/lib/python3.12/site-packages/material/plugins/social/plugin.py", line 876, in _fetch_font_from_google_fonts
    archive = ZipFile(f)
              ^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.2/x64/lib/python3.12/zipfile/__init__.py", line 1341, in __init__
    self._RealGetContents()
  File "/opt/hostedtoolcache/Python/3.12.2/x64/lib/python3.12/zipfile/__init__.py", line 1408, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
Error: Process completed with exit code 1.
```

Looks like the social plugin? Investigating...

Thanks again.

**Reply by kamilkrzyskow** [reply]
Could be related:
- https://github.com/squidfunk/mkdocs-material/discussions/5761

The font is not available anymore 🧐?

**Reply by tbedford** [reply]
Interesting. Looked very similar. I tried changing the font to Roboto, but it generated the same error. And prior to this morning everything worked fine. I am suspecting social plugin somehow? I've removed it for now and everything builds OK.

**Reply by kamilkrzyskow** [reply]
Since Roboto doesn't work either Google Fonts doesn't work or the API has changed and the Social plugin doesn't download the fonts zip properly anymore.

**Comment by shepherdjerred** [comment]
I started seeing the same issue today.

```
   ./docs/main+build *failed* |   File "/usr/lib/python3.10/concurrent/futures/thread.py", line 58, in run
   ./docs/main+build *failed* |     result = self.fn(*self.args, **self.kwargs)
   ./docs/main+build *failed* |   File "/usr/local/lib/python3.10/dist-packages/material/plugins/social/plugin.py", line 444, in _render
   ./docs/main+build *failed* |     image = self._render_typography(layer, image)
   ./docs/main+build *failed* |   File "/usr/local/lib/python3.10/dist-packages/material/plugins/social/plugin.py", line 521, in _render_typography
   ./docs/main+build *failed* |     path = self._resolve_font(family, styles)
   ./docs/main+build *failed* |   File "/usr/local/lib/python3.10/dist-packages/material/plugins/social/plugin.py", line 808, in _resolve_font
   ./docs/main+build *failed* |     self._fetch_font_from_google_fonts(family)
   ./docs/main+build *failed* |   File "/usr/local/lib/python3.10/dist-packages/material/plugins/social/plugin.py", line 854, in _fetch_font_from_google_fonts
   ./docs/main+build *failed* |     archive = ZipFile(f)
   ./docs/main+build *failed* |   File "/usr/lib/python3.10/zipfile.py", line 1269, in __init__
   ./docs/main+build *failed* |     self._RealGetContents()
   ./docs/main+build *failed* |   File "/usr/lib/python3.10/zipfile.py", line 1336, in _RealGetContents
   ./docs/main+build *failed* |     raise BadZipFile("File is not a zip file")
   ./docs/main+build *failed* | zipfile.BadZipFile: File is not a zip file
```

**Comment by alexvoss** [comment]
This might explain it? https://github.com/squidfunk/mkdocs-material/issues/6983

Is it possible that local builds succeed because they have fonts cached?

**Reply by tbedford** [reply]
Yes, you are spot on. I was checking the `.cache` just now and the fonts are all there.

**Reply by alexvoss** [reply]
This might provide a way forward in the short-term if you can stage them into your build? As far as I can see, Google have thrown a real curveball here and this will require some changes to the social plugin that are not just one-liners. I do wonder where the default fonts come from, are they also from GF? Gotta go and test that.

**Edit**: yes, it seems the default fonts also want to come from GF, so just reverting to them is not a solution.

**Reply by alexvoss** [reply]
I posted a workaround at: https://github.com/squidfunk/mkdocs-material/issues/6983

Also created a PR for what might be a fix, though one that requires manually downloading the fonts.

**Comment by squidfunk** [comment]
The problem has been fixed in both editions – please upgrade to `master` and report whether this fixes your issue. We need some feedback because of the many ways the social plugin can be configured before we issue a bugfix release.

For more information, see https://github.com/squidfunk/mkdocs-material/issues/6983#issuecomment-2028580845.
